k8s/upgrades/0.8.1-0.8.2/jiva_volume_upgrade.sh

```diff
@@ -0,0 +1,227 @@
+# NOTES: Placeholder for resourcename in the patch files are   #
+# replaced with respective values derived from the PV in the   #
+# previous step                                                #
+################################################################
+
+# Check if openebs resources exist and provisioned version is 0.8
+
+if [[ -z $c_rs ]]; then
+    echo "Target Replica set not found"; exit 1;
+fi
+
+if [[ -z $r_rs ]]; then
+    echo "Replica Replica set not found"; exit 1;
+fi
+
+if [[ -z $c_dep ]]; then
+    echo "Target deployment not found"; exit 1;
+fi
+
+if [[ -z $r_dep ]]; then
+    echo "Replica deployment not found"; exit 1;
+fi
+
+if [[ -z $c_svc ]]; then
+    echo "Target service not found"; exit 1;
+fi
+
+if [[ -z $r_name ]]; then
+    echo "Replica container not found"; exit 1;
+fi
+
+if [[ -z $c_name ]]; then
+    echo "Target container not found"; exit 1;
+fi
+
+controller_version=`kubectl get deployment $c_dep -n $ns -o jsonpath='{.metadata.labels.openebs\.io/version}'`
+if [[ "$controller_version" != "$current_version" ]] && [[ "$controller_version" != "$target_upgrade_version" ]]; then
+    echo "Current Target deployment $c_dep version is not $current_version or $target_upgrade_version";exit 1;
+fi
+replica_version=`kubectl get deployment $r_dep -n $ns -o jsonpath='{.metadata.labels.openebs\.io/version}'`
+if [[ "$replica_version" != "$current_version" ]] && [[ "$replica_version" != "$target_upgrade_version" ]]; then
+    echo "Current Replica deployment $r_dep version is not $current_version or $target_upgrade_version";exit 1;
+fi
+
+controller_svc_version=`kubectl get svc $c_svc -n $ns -o jsonpath='{.metadata.labels.openebs\.io/version}'`
+if [[ "$controller_svc_version" != $current_version ]] && [[ "$controller_svc_version" != "$target_upgrade_version" ]] ; then
+    echo "Current Target service $c_svc version is not $current_version or $target_upgrade_version";exit 1;
+fi
+
+# Get the number of replicas configured.
+# This field is currently not used, but can add additional validations
+# based on the nodes and expected number of replicas
+rep_count=`kubectl get deploy $r_dep --namespace $ns -o jsonpath="{.spec.replicas}"`
+
+# Get the list of nodes where replica pods are running, delimited by ':'
+rep_nodenames=`kubectl get pods -n $ns \
+ -l "openebs.io/persistent-volume=$pv" -l "openebs.io/replica=jiva-replica" \
+ -o jsonpath="{range .items[*]}{@.spec.nodeName}:{end}"`
+
+echo "Checking if the node with replica pod has been labeled with $replica_node_label"
+for rep_node in `echo $rep_nodenames | tr ":" " "`; do
+    nl="";nl=`kubectl get nodes $rep_node -o jsonpath="{.metadata.labels.openebs-pv-$pv}"`
+    if [  -z "$nl" ];
+    then
+       echo "Labeling $rep_node";
+       kubectl label node $rep_node "openebs-pv-${pv}=$replica_node_label"
+    fi
+done
+
+
+sed -u "s/@r_name@/$r_name/g" | sed -u "s/@target_version@/$target_upgrade_version/g"  > jiva-replica-patch.json
```

Comment:
file name is missing.. also, @c_name@ is also missing

Reply:
Done